src/preset.ts

```diff
@@ -1130,16 +1130,20 @@ export async function expandConfigurePreset(folder: string, name: string, worksp
  * calling configurePresets() or userConfigurePresets(). Getting the presets plus included map is useful on Select Preset when we want to be able to
  * apply the Vs Dev Env to the preset and want the entire list of unexpanded presets, including the inlcuded presets.
  */
-async function expandConfigurePresetImpl(folder: string, name: string, allowUserPreset: boolean = false, usePresetsPlusIncluded: boolean = false, errorHandler?: ExpansionErrorHandler): Promise<ConfigurePreset | null> {
-    let preset = getPresetByName(configurePresets(folder, usePresetsPlusIncluded), name);
-    if (preset) {
-        return expandConfigurePresetHelper(folder, preset, false, usePresetsPlusIncluded, errorHandler);
-    }
-
-    if (allowUserPreset) {
-        preset = getPresetByName(userConfigurePresets(folder, usePresetsPlusIncluded), name);
+async function expandConfigurePresetImpl(folder: string, name: string, allowUserPreset: boolean = false, usePresetsPlusIncluded: boolean = false, errorHandler?: ExpansionErrorHandler, inheritedByPreset?: ConfigurePreset): Promise<ConfigurePreset | null> {
+    const presetList = inheritedByPreset ? inheritedByPreset.__file!.configurePresets : configurePresets(folder, usePresetsPlusIncluded);
+    const validInherit = presetList !== undefined && presetList.filter(p => p.name === name).length > 0;
+    if (validInherit) {
+        let preset = getPresetByName(configurePresets(folder, usePresetsPlusIncluded), name);
```

Comment:
Do we need to use `presetList` here on line 1137 rather than configurePresets again? 

I guess it would be fine either way, because we've already validated? 

THis is true for all other methods as well

Reply:
yes since the validation has already happened, we can call regular configurepresets. I'm calling the configurepresets here, to make sure we are getting the right/updated preset based on the usepresetsplusincluded